src/backend/utils/init/miscinit.c

```diff
@@ -1693,6 +1694,22 @@ process_shared_preload_libraries(void)
 				   false);
 
 	process_shared_preload_libraries_in_progress = false;
+
+	{
+		struct stat st;
+
+		if (file_read_buffer_modify_hook == NULL)
+		{
+			int file_exists = (stat("data_encryption.key", &st) ? (errno != ENOENT) : true);
```

Comment:
What about the following logic? If we can stat the file or the errno not equals to ENOENT, we assume the file exists.
```c
int file_exists = state("data_encryption.key", &st) == 0 || errno != ENOENT;
```